src/restic/checker/checker.go

```diff
@@ -587,10 +587,6 @@ func (c *Checker) checkTree(id restic.ID, tree *restic.Tree) (errs []error) {
 	for _, node := range tree.Nodes {
 		switch node.Type {
 		case "file":
-			if node.Content == nil {
```

Comment:
This check was introduced to recognize an error condition where the archiver won't properly save a list of blobs, but rather leave the `.Content` set to `nil`. I would have left the check, even if it seems redundant.